libs/ngx-dynamic-form-builder/src/lib/utils/dynamic-form-group.ts

```diff
@@ -406,115 +143,498 @@ export class DynamicFormGroup<TModel> extends FormGroup {
+
+            // Add the control to the FormArray
+            formArray.controls.push(newFormControl);
           }
         }
       }
+
+      // Handle Control
+      else {
+        const newObject = this._object ? this._object[key] : [];
+        this.controls[key].setValue(this._object && newObject ? newObject : undefined);
+      }
     });
   }
-  validateAllFormFields() {
-    Object.keys(this.controls).forEach(field => {
-      const control = this.get(field);
-      if (control instanceof FormControl) {
-        control.markAsTouched({ onlySelf: true });
-      } else {
-        if (control instanceof DynamicFormGroup) {
-          control.validateAllFormFields();
-        } else {
-          if (control instanceof FormArray) {
-            for (let i = 0; i < (control as FormArray).controls.length; i++) {
-              if ((control as FormArray).controls[i] instanceof FormControl) {
-                ((control as FormArray).controls[i] as FormControl).markAsTouched({ onlySelf: true });
-              } else {
-                if ((control as FormArray).controls[i] instanceof DynamicFormGroup) {
-                  ((control as FormArray).controls[i] as DynamicFormGroup<any>).validateAllFormFields();
+}
+
+
+export function getClassValidators<TModel>(factoryModel: ClassType<TModel>, fields: Dictionary, validatorOptions?: ValidatorOptions) {
+
+  // Get the validation rules from the object decorators
+  const allValidationMetadatas: ValidationMetadata[] = getFromContainer(MetadataStorage).getTargetValidationMetadatas(factoryModel, '');
+
+  // Get the validation rules for the validation group: https://github.com/typestack/class-validator#validation-groups
+  const validationGroupMetadatas: ValidationMetadata[] = getFromContainer(MetadataStorage).getTargetValidationMetadatas(
+      factoryModel,
+      '',
+      validatorOptions && validatorOptions.groups ? validatorOptions.groups : undefined
+    );
+
+  const formGroupFields = {};
+  const validator = new Validator();
+
+  // Loop through all fields in the form definition
+  Object.keys(fields).filter(key => key.indexOf('__') !== 0).forEach(fieldName => {
+
+    // Conditional Validation for the field
+    const conditionalValidations: ValidationMetadata[] = [];
+    validationGroupMetadatas.forEach(validationMetadata => {
+      if (isPropertyValidatorOfType(validationMetadata, fieldName, ValidationKeys.conditional.type)) {
+        conditionalValidations.push(validationMetadata);
+      }
+    });
+
+    // All Nested Validation for the field
+    const allNestedValidations: ValidationMetadata[] = [];
+      allValidationMetadatas.forEach(validationMetadata => {
+        if (isPropertyValidatorOfType(validationMetadata, fieldName, ValidationKeys.nested.type)) {
+          allNestedValidations.push(validationMetadata);
+        }
+      });
+
+    // Nested Validation for the field for the requested class-validator group
+    const nestedGroupValidations: ValidationMetadata[] = [];
```

Comment:
Renamed to call out it's use with https://github.com/typestack/class-validator#validation-groups